File: coinsaver-api/src/main/java/com/coinsaver/api/dtos/response/UpdateTransactionResponseDto.java
package com.coinsaver.api.dtos.response;

import com.coinsaver.core.enums.StatusType;
import com.coinsaver.core.enums.TransactionCategoryType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class UpdateTransactionResponseDto {

    private Long id;
    private BigDecimal amount;
    private LocalDate payDay;
    private String description;
    private StatusType status;
    private TransactionCategoryType category;
    private Boolean fixedExpense;
    private Boolean paid;
    private Integer repeat;
}

File: coinsaver-front/src/app/dtos/transactions/response/update-transaction.response.dto.ts
import { TransactionCategoryTypeEnum } from 'src/app/enums/transaction-category-type.enum';
import { StatusTypeEnum } from '../../../enums/status-type.enum';

export class UpdateTransactionResponseDto {
  constructor(
    public id: number,

    public amount: number,

    public payDay: string,

    public description: string,

    public status: StatusTypeEnum,
    public category: TransactionCategoryTypeEnum,

    public fixedExpense: boolean,

    public paid: boolean,

    public repeat: number,
  ) {}
}
